### PR TITLE
Release v2.0.0.1

### PR DESCRIPTION
**Pull request by BenjaminRueth**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,239 @@
...
+
+We recommend [using pip3](https://github.com/precice/precice/blob/develop/src/precice/bindings/python/README.md#using-pip3) for the sake of simplicity.
+
+## Using pip3
+
+### preCICE system installs
+
+For system installs of preCICE, this works out of the box.
+
+You can either install from PyPI:
```
Comment (fsimonis) [inline README.md]: What about using pip to download and install from the github repo as an archive?

File: README.md [hunk 2]
```diff
@@ -1,2 +1,239 @@
...
+```
+wget -q https://bootstrap.pypa.io/get-pip.py -O get-pip.py && python3 get-pip.py
+```
+*Be aware that `python3 get-pip.py` might require root privileges.*
+
+Check your version of pip via `pip3 --version`. For version 8.1.1 and 9.0.1 we know that this problem occurs. *Remark:* you get versions 8.1.1 of pip if you use `sudo apt install python3-pip` on Ubuntu 16.04 (pip version 9.0.1 on Ubuntu 18.04)
+
+### I'm using preCICE < 2.0.0, but there is no matching version of the bindings. What can I do?
+
+If you want to use the old python bindings (released with preCICE version < 2.0.0), please refer to the documentation of the corresponding preCICE version.
```
Comment (fsimonis) [inline README.md]:
```suggestion
If you want to use the old experimental python bindings (released with preCICE version < 2.0.0), please refer to the documentation of the corresponding preCICE version. 
```

File: README.md [hunk 3]
```diff
@@ -1,2 +1,239 @@
...
+
+Check your version of pip via `pip3 --version`. For version 8.1.1 and 9.0.1 we know that this problem occurs. *Remark:* you get versions 8.1.1 of pip if you use `sudo apt install python3-pip` on Ubuntu 16.04 (pip version 9.0.1 on Ubuntu 18.04)
+
+### I'm using preCICE < 2.0.0, but there is no matching version of the bindings. What can I do?
+
+If you want to use the old python bindings (released with preCICE version < 2.0.0), please refer to the documentation of the corresponding preCICE version. 
+
+### Installing the python bindings for Python 2.7.17
+
+*Note that the instructions in this section are outdated and refer to the deprecated python bindings. Until we have updated information on the installation procedure for the python bindings under this use-case, we will keep these instructions, since they might still be very useful* (Originally contributed by [huangq1234553](https://github.com/huangq1234553) to the precice wiki in [`precice/precice/wiki:8bb74b7`](https://github.com/precice/precice/wiki/Dependencies/8bb74b78a7ebc54983f4822af82fb3d638021faa).)
```
Comment (fsimonis) [inline README.md]: You  could put this into:

<details><summary>Click to travel back in time.</summary>
Hello
</details>

File: setup.py [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+# this version number may be increased, if changes for the bindings are required
+bindings_version = version.Version("1")
+APPVERSION = version.Version(str(precice_version) + "." + str(bindings_version))
+
+PYTHON_BINDINGS_PATH = os.path.dirname(os.path.abspath(__file__))
+
+def get_extensions(is_test):
+    compile_args = []
+    link_args = []
+    compile_args.append("-Wall")
```
Comment (fsimonis) [inline setup.py]: Is the warning flag necessary? If the local `CXX_FLAGS` are used for compiling, then we can simply drop this.

File: setup.py [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+class my_test(test, object):
+    def initialize_options(self):
+        self.distribution.is_test = True       
+        super().initialize_options()
+
+# build precice.so python extension to be added to "PYTHONPATH" later
+setup(
+    name=APPNAME,
+    version=str(APPVERSION),
+    description='Python language bindings for preCICE coupling library',
```
Comment (fsimonis) [inline setup.py]:
```suggestion
    description='Python language bindings for the preCICE coupling library',
```

File: setup.py [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+    author='the preCICE developers',
+    author_email='info@precice.org',
+    license='LGPL-3.0',
+    python_requires='>=3',
+    install_requires=['numpy', 'mpi4py'],  # mpi4py is only needed, if preCICE was compiled with MPI, see https://github.com/precice/python-bindings/issues/8
+    cmdclass={'test': my_test,
+              'build_ext': my_build_ext,
+              'build': my_build,
+              'install': my_install},
+    #ensure pxd-files:
```
Comment (fsimonis) [inline setup.py]:
```suggestion
```

File: solverdummy/.gitignore [hunk 7]
```diff
@@ -0,0 +1,4 @@
+*events.json
+venv
+.idea
+*.log
```
Comment (fsimonis) [inline solverdummy/.gitignore]:
```suggestion
*.log
__pycache__
```

File: solverdummy/precice-config.xml [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      <min-iteration-convergence-measure min-iterations="5" data="Forces" mesh="MeshOne"/>
+      <exchange data="Forces"     mesh="MeshOne" from="SolverOne" to="SolverTwo" />
+      <exchange data="Velocities" mesh="MeshOne" from="SolverTwo" to="SolverOne"/>
+    </coupling-scheme:serial-implicit>                           
+             
+    <!--
+    <coupling-scheme:serial-explicit> 
+      <participants first="SolverOne" second="SolverTwo" /> 
+      <max-timesteps value="10" />
+      <timestep-length value="1.0" />
```
Comment (fsimonis) [inline solverdummy/precice-config.xml]:
```suggestion
      <time-window-size value="1.0" />
```

File: solverdummy/precice-config.xml [hunk 9]
```diff
@@ -0,0 +1,62 @@
...
+      <max-iterations value="3" />
+      <min-iteration-convergence-measure min-iterations="5" data="Forces" mesh="MeshOne"/>
+      <exchange data="Forces"     mesh="MeshOne" from="SolverOne" to="SolverTwo" />
+      <exchange data="Velocities" mesh="MeshOne" from="SolverTwo" to="SolverOne"/>
+    </coupling-scheme:serial-implicit>                           
+             
+    <!--
+    <coupling-scheme:serial-explicit> 
+      <participants first="SolverOne" second="SolverTwo" /> 
+      <max-timesteps value="10" />
```
Comment (fsimonis) [inline solverdummy/precice-config.xml]:
```suggestion
      <max-time-windows value="10" />
```

File: solverdummy/requirements.txt [hunk 10]
```diff
@@ -0,0 +1,3 @@
+pyprecice>=0.1
```
Comment (fsimonis) [inline solverdummy/requirements.txt]:
```suggestion
pyprecice==2.0
```

Matches all adapter version based preCICE 2.0 with arbitrary patch and adapter version.

File: README.md (hunk 3 above)
Comment (BenjaminRueth) [reply README.md]: Done. See 3be9aaf

File: README.md (hunk 1 above)
Comment (BenjaminRueth) [reply README.md]: Done e440e7c

File: setup.py (hunk 4 above)
Comment (BenjaminRueth) [reply setup.py]: I removed this line and `-Wall` still shows up in the compilation command:

```
x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -I/usr/include/python3.6m -c /home/benjamin/python-bindings/test/test_bindings_module.cpp -o build/temp.linux-x86_64-3.6/home/benjamin/python-bindings/test/test_bindings_module.o -std=c++11 -I/home/benjamin/.local/lib/python3.6/site-packages/numpy/core/include
```

**Pull request merged**